ompi/mca/hook/comm_method/hook_comm_method_fns.c

```diff
@@ -473,73 +502,24 @@ ompi_report_comm_methods(int called_from_location) // 1 = from init, 2 = from fi
 // Each host leader fills in a "numhosts" sized array method[] of
 // how it communicates with each peer.
     for (i=0; i<nleaderranks; ++i) {
```

Comment:
`mymethod`'s size is based on `numhosts` in the malloc above on line 409, not `nleaderranks`. I assume that `nleaderranks` should be <= `numhosts`, but could we make this consistent?

Reply:
(pushing another commit, intend to squash once confirmed it's what we want)

Yeah, they were equivalent, so I removed numhosts and switched them all to nleaderranks